android/src/main/java/com/brentvatne/exoplayer/ReactExoplayerView.java

```diff
@@ -1706,54 +1705,42 @@ public void onCues(CueGroup cueGroup) {
 
     // ReactExoplayerViewManager public api
 
-    public void setSrc(final Uri uri, final long startPositionMs, final long cropStartMs, final long cropEndMs, final String extension, Map<String, String> headers, MediaMetadata customMetadata) {
-
-        if (!Util.areEqual(this.customMetadata, customMetadata) && player != null) {
-            MediaItem currentMediaItem = player.getCurrentMediaItem();
-
-            if (currentMediaItem != null) {
-
-                MediaItem newMediaItem = currentMediaItem.buildUpon().setMediaMetadata(customMetadata).build();
-
-                // This will cause video blink/reload but won't louse progress
-                player.setMediaItem(newMediaItem, false);
-            }
-        }
-
-        if (uri != null) {
-            boolean isSourceEqual = uri.equals(srcUri) && cropStartMs == this.cropStartMs && cropEndMs == this.cropEndMs;
+    public void setSrc(Source source) {
+        if (source.getUri() != null) {
+            clearResumePosition();
+            boolean isSourceEqual = source.isEquals(this.source);
             hasDrmFailed = false;
-            this.srcUri = uri;
-            this.startPositionMs = startPositionMs;
-            this.cropStartMs = cropStartMs;
-            this.cropEndMs = cropEndMs;
-            this.extension = extension;
-            this.requestHeaders = headers;
+            this.source = source;
             this.mediaDataSourceFactory =
                     DataSourceUtil.getDefaultDataSourceFactory(this.themedReactContext, bandwidthMeter,
-                            this.requestHeaders);
-            this.customMetadata = customMetadata;
+                            source.getHeaders());
 
+            // refresh custom Metadata
+            customMetadata = ConfigurationUtils.buildCustomMetadata(source.getMetadata());
```

Comment:
https://github.com/TheWidlarzGroup/react-native-video/pull/3835/files#diff-a35ae7e6ae6efee3b22522a836bcbcdd9d59a214f087df454e4ccbb8cd5c0e76L1711
this compare code is missing

Reply:
I don't think this is really mandatory, BTW, I add this check again and test the behavior. Looks good !
I can gain few ms in corner case

Reply:
In my production environment
https://github.com/TheWidlarzGroup/react-native-video/issues/3763
Without this code, I was experiencing the above issue.🥲
I haven't found the root cause, but it appears to be a race condition caused by the re-render.

Reply:
@YangJonghun can It the DRM issue I fixed ? https://github.com/TheWidlarzGroup/react-native-video/issues/3082

Reply:
I understand that you're asking me to reproduce that issue, and It, hard to reproducing it because I don't know exactly how to reproduce it or what's causing it (please let me know if I'm misunderstanding you)

Reply:
No no, I just highlight you the DRM issue ;) 
I put back the test by the way !

Reply:
Thanks :)